File: 447. Number of Boomerangs/Java/Number of Boomerangs.java
public class Solution {
    public int numberOfBoomerangs(int[][] points) {
        int result = 0;
        for (int[] pointA : points) {
            Map<Integer, Integer> distances = new HashMap<>();
            for (int[] pointB : points) {
                int distance = (pointA[0] - pointB[0]) * (pointA[0] - pointB[0]) + (pointA[1] - pointB[1]) * (pointA[1] - pointB[1]);
                if (!distances.containsKey(distance)) {
                    distances.put(distance, 0);
                }
                distances.put(distance, distances.get(distance) + 1);
            }
            for (int item : distances.values()) {
                result += item * (item - 1);
            }
        }
        return result;
    }
}

File: 447. Number of Boomerangs/C++/Number of Boomerangs.cpp
class Solution {
public:
    int numberOfBoomerangs(vector <pair<int, int>> &points) {
        int result = 0;
        for (auto &pointA:points) {
            map<int, int> distances;
            for (auto &pointB:points) {
                int distance = (pointA.first - pointB.first) * (pointA.first - pointB.first) +
                               (pointA.second - pointB.second) * (pointA.second - pointB.second);
                if (distances.find(distance) == distances.end()) {
                    distances[distance] = 0;
                }
                distances[distance]++;
            }
            for (auto &item:distances) {
                result += item.second * (item.second - 1);
            }
        }
        return result;
    }
};